platform.md

```diff
@@ -880,16 +878,11 @@ For more information on build reproducibility see [https://reproducible-builds.o
 #### `analyzed.toml` (TOML)
 
 ```toml
-[previous-image]
-  reference = "<image reference>"
-[previous-image.metadata]
-  # previous image layer metadata
-[run-image]
+[image]
   reference = "<image reference>"
-  mixins = ["libgc", "libpq"]
```

Comment:
Feels like we should do one of:

1. Keep run-image in `analyzed.toml` and remove it as an input to exporter. Therefore we ensure that the image we are validating read access to, is in fact the one that gets used during export.

2. Remove run-image as an input to analyzer. This keeps the dynamic run-image selection option open.

Reply:
I think option 2 makes the most sense to me.

The dynamic run-image selection mentioned in another RFC has the builder contain an ordered list of run images. If we go in the direction of having multiple run images defined in the builder, we could introduce all possible run images into `analyzed.toml` so we have the exact image refs and any metadata (PURL list) available for later phases and validate access. Maybe not having access to a specific run image option isn't a failure mode, but limits the choices when we get to selection process.

Reply:
Would this mean we also remove `-stack` as an input to the analyzer?

Reply:
After thinking about this a bit more... is there great harm in leaving things as they are now? Since checking read access for the run image is just a convenience and will not cause "wrong" things to happen if the run image provided to analyzer is different from the run image provided to exporter.

Reply:
I like option 1 although knowing the lifecycle code, it is more work to implement. And to Jesse's point - in a world where we are doing dynamic run image selection, does a `-run-image` flag still make sense for analyzer? That might influence what we decide here.

Reply:
Does doing 1 make switching that much harder?

Reply:
To summarize what I think we arrived at - 

We'd like to
* fail fast whenever possible (so validate read access for the run image during analyze)
* lock the run image to a reference, to avoid introducing bugs in the future if we assume that run image provided to analyzer & run image provided to exporter are guaranteed to be the same
* bundle breaking changes together

All this points to making the schema addition in analyzed.toml and removing the run image input to exporter.

Reply:
Just to confirm, `-stack` will still be an input to the exporter - it just won't be used to derive the run image, correct?

Reply:
Please see 7108fae

Reply:
Should analyzed.toml now be **required** for exporter? The code currently treats it as optional...

Reply:
> Should analyzed.toml now be required for exporter? The code currently treats it as optional...

Yes I think it should be required now. Previously, was only theoretically optional if there was no previous image (even though we still wrote and passed through an empty analyzed.toml iifc).